### PR TITLE
GITLAB-779 Add original branch when building branches

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

**Review by rouazana — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

File: build-github-branch [hunk 1]
```diff
@@ -5,11 +5,12 @@ class BranchOnRepo {
     String repo
     String branch
     String version
+    String originalBranch = ""
 }
 
 def branchesToBuild = [
     new BranchOnRepo('apache/james-project', 'master', '3.2.0-SNAPSHOT'),
-    new BranchOnRepo('apache/james-project', '3.1.0', '3.1.0')
+    new BranchOnRepo('apache/james-project', '3.1.x', '3.1.0', '3.1.0')
```
Comment (chibenwa) [inline build-github-branch:13]: Isn't it `3.1.0` as base branch too?

File: build-github-branch [hunk 2]
```diff
@@ -5,11 +5,12 @@ class BranchOnRepo {
     String repo
     String branch
     String version
+    String originalBranch = ""
 }
 
 def branchesToBuild = [
     new BranchOnRepo('apache/james-project', 'master', '3.2.0-SNAPSHOT'),
```
Comment (chibenwa) [inline build-github-branch:12]: No explicit base branch parameter here?

Comment (aduprat) [reply build-github-branch:12]: No, the default is `master`

File: build-github-branch (hunk 1 above)
Comment (aduprat) [reply build-github-branch:13]: I was thinking on that, and I think we only want to build _living_ branches.
`3.1.0` will never change, development should be merged on `3.1.x`

**Review by chibenwa — APPROVED** [review]

**Pull request merged**